### PR TITLE
Gamma cyc

### PR DESCRIPTION
**Pull request by marjanfamili**

### CODE REVIEW

**Comment by marjanfamili** [comment]
#118 #116

File: avae/loss.py [hunk 1]
```diff
@@ -90,24 +90,24 @@ class AVAELoss:
 
     """
 
-    def __init__(self, device, beta, gamma=1, lookup_aff=None, recon_fn="MSE"):
+    def __init__(self, device, beta, gamma, lookup_aff=None, recon_fn="MSE"):
```
Comment (jolaem) [inline avae/loss.py:93]: Will this not break if there's no gamma?

Comment (marjanfamili) [reply avae/loss.py:93]: Do you mean no gamma in the input file ?

File: avae/train.py [hunk 2]
```diff
@@ -82,22 +86,46 @@ def train(
         params=vae.parameters(), lr=learning  # , weight_decay=1e-5
     )
 
-    beta_arr = KLD_weight(
+    print(epochs, cyc_method_beta, beta_min, beta_max, beta_cycle, beta_ratio)
```
Comment (jolaem) [inline avae/train.py:89]: Is this a leftover debug print statement or useful feedback? If latter, can we add some text to explain what's being printed?

File: avae/loss.py (hunk 1 above)
Comment (jolaem) [reply avae/loss.py:93]: Ah, because it's an array now, gotcha. I was just confused as gamma was optional as you don't need it to run vanilla beta-VAE.

**Review by jolaem — CHANGES_REQUESTED** [review]
This fails for me with:

`Traceback (most recent call last):
  File "/home/jola/mygit/affinity-vae/run.py", line 593, in <module>
    run()
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/jola/mygit/affinity-vae/run.py", line 489, in run
    data[key] = config.DEFAULT_RUN_CONFIGS[key]
KeyError: 'kl_weight_method'
`

**Review by jolaem — CHANGES_REQUESTED** [review]
This fails for me with:

`Traceback (most recent call last):
  File "/home/jola/mygit/affinity-vae/run.py", line 593, in <module>
    run()
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/jola/mygit/affinity-vae/run.py", line 489, in run
    data[key] = config.DEFAULT_RUN_CONFIGS[key]
KeyError: 'kl_weight_method'
`

**Comment by marjanfamili** [comment]
The key has changed so you need to update your input

File: avae/train.py [hunk 3]
```diff
@@ -82,22 +86,46 @@ def train(
...
+    gamma_arr = cyc_annealing(
+        epochs,
+        cyc_method_gamma,
+        start=gamma_min,
+        stop=gamma_max,
+        n_cycle=gamma_cycle,
+        ratio=gamma_ratio,
+    ).var
 
+    if config.VIS_BET:
```
[the remark this answers is not in the document]
Comment (marjanfamili) [reply avae/train.py:121]: ok

File: avae/train.py (hunk 2 above)
Comment (marjanfamili) [reply avae/train.py:89]: thanks for pointing this out, I always miss one :))

**Comment by marjanfamili** [comment]
The last commit addresses #121

**Review by jolaem — APPROVED** [review]

**Pull request merged**